spring-graphql-docs/src/docs/asciidoc/index.adoc

```diff
@@ -480,6 +480,57 @@ assertThat(books.get(0).getName()).isEqualTo("...");
+Instead, Spring for GraphQL is an API gateway that leverages existing Spring technology
+following common programming models to expose underlying data sources through GraphQL.
+
+Domain-driven design is the suggested approach to manage complexity  when using Spring Data.
+So by design, a GraphQL API built on top of Spring Data can leverage only what's already
+provided by an application.
+It, therefore, must adhere to the constraints of an aggregate.
+By definition, an aggregate is only valid if it is loaded in its entirety.
+Partially loaded aggregates may impose a limitation on aggregate functionality.
+
+With Spring Data you can chose whether you want to let your aggregate participate as
+underlying data model directly to be exposed as GraphQL result or whether you want to
+apply projections to your data model before returning it as GraphQL query result.
+
+The advantage of using aggregates is that you do not require additional code to expose
+data through repositories. When processing a query, the integration layer transforms the
+query selection into property paths. It provides these hints of which properties to
+materialize to the underlying Spring Data module that limits the field (or column) selection.
+
+Sometimes, an already reduced set of fields can be useful when exposing data. Also, some
+arrangements might require transformations to be applied before data is returned for a
+GraphQL query. Spring Data supports for these scenarios projections: Interface and DTO
+Projections.
+
+Interface projections define a fixed set of properties to expose. Properties may or may
+not be `null`, depending on the query result. A plain, https://docs.spring.io/spring-data/commons/docs/current/reference/html/#projections.interfaces.closed[closed interface projection]
+can be useful if you cannot partially materialize the aggregate object but you still
+want to expose a subset of properties.
+
+You can use interface projections to apply a lightweight set of data transformations,
+such as concatenations, computations or applying a static function to a property.
+
+https://docs.spring.io/spring-data/commons/docs/current/reference/html/#projections.interfaces.open[Open projections]
+leverage Spring's `@Value` annotation and
+{spring-framework-ref-docs}/core.html#expressions[SpEL expressions]for transformations.
+
+In both cases, interface projections define which properties to load from the underlying
+data source.
+
+DTO projections offer the highest possible level of customization as you can place your
+transformation code either in the constructor or the getter method.
+
+DTO projections materialize from a query where the individual projections are
+determined by the projection itself. DTO projections are commonly used with full-args
+constructors (e.g. Java Records) and therefore they can be only constructed if all
+required fields (or columns) are part of the query result.
```

Comment:
```suggestion
DTO projections materialize from a GraphQL operation, where the individual projections are
determined by the projection itself. DTO projections are commonly used with full-args
constructors (e.g. Java records) and therefore they can only be constructed if all
required fields (or columns) are part of the operation result.
```
This paragraph reads a bit confusingly for me, what does it mean exactly?
`where the individual projections are determined by the projection itself`
`required fields (or columns) are part of the operation/query result` - should this be the operation's selection set?

Reply:
query is here the database query. 

`individual projections are` should be `individual properties are`